py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,309 @@
+import argparse
+import json
+import logging
+import os
+import urllib.parse
+from typing import Any, Mapping, Optional, Sequence, TextIO, Tuple, TypedDict
+from urllib.request import Request, urlopen
+
+from usageaccountant.accumulator import UsageAccumulator, UsageUnit
+
+headers = {
+    "DD-APPLICATION-KEY": os.environ.get("DATADOG_APP_KEY", ""),
+    "DD-API-KEY": os.environ.get("DATADOG_API_KEY", ""),
+}
+
+logger = logging.getLogger("fetcher")
+
+# TODO remove handlers before prod
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.FileHandler("debug.log"), logging.StreamHandler()],
+)
+
+
+class DatadogResponseUnit(TypedDict):
+    plural: str
+    name: str
+
+
+class DatadogResponseSeries(TypedDict):
+    unit: Sequence[DatadogResponseUnit]
```

Comment:
Why is the unit a sequence of `DatadogResponseUnit`. Logically speaking a single time series should have only one unit or am I wrong ?

Reply:
It is indeed a list because of how they do units in case of something like a rate, bytes per second, wherein bytes and second are two different unit objects.
[Reference](https://datadog-api-client.readthedocs.io/en/latest/datadog_api_client.v1.model.html#datadog_api_client.v1.model.metrics_query_metadata.MetricsQueryMetadata).
Need to think through how we handle units altogether. I'm modifying code accept units via the query file.

Reply:
Then if they are two different object this is a `Tuple[DatadogResponseUnit, DatadogResponseUnit]`. Sequence is unbounded.